.github/workflows/publish.yml

```diff
@@ -1,107 +1,92 @@
+      - name: upload-sdist-artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.build-package.outputs.sdist-package-name }}
+          path: dist/
+          if-no-files-found: error
 
-      - name: Get current pushed tag
-        run:  |
-          echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
-          echo "$RELEASE_VERSION"
-          
-      #----------------------------------------------
-      #    override version tag 
-      #----------------------------------------------      
-      - name: Override version tag
-        run: poetry run python3 override_version.py
-        shell: sh
+      - name: upload-bdist-artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.build-package.outputs.bdist-package-name }}
+          path: dist/
+          if-no-files-found: error
 
-      #----------------------------------------------
-      #    publish to testpypi
-      #----------------------------------------------      
-      # - run: poetry config repositories.testpypi https://test.pypi.org/legacy/
-      # - run: poetry config pypi-token.testpypi ${{ secrets.TWINE_TEST_TOKEN }}
-      # - name: Publish package to test Pypi
-      #   run: poetry publish -vvvv --build -r testpypi
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ github.event.release.prerelease == true && 'testpypi' || 'pypi' }}
+      url: ${{ github.event.release.prerelease == true && 'https://test.pypi.org/p/schematicpy' || 'https://pypi.org/p/schematicpy' }}
+    permissions:
+      id-token: write
+    steps:
+      - name: download-sdist
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ needs.build.outputs.sdist-package-name }}
+          path: dist
 
-      #----------------------------------------------
-      #    check tag
-      #----------------------------------------------      
-      - name: Check Tag
-        id: check-tag
-        run: |
-          if [[ ${{ github.event.ref }} =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-              echo ::set-output name=match::true
-          fi
-      #----------------------------------------------
-      #    publish to pypi
-      #----------------------------------------------  
-      - name: Publish package to Pypi
-        id: publish-to-pypi
-        if: steps.check-tag.outputs.match == 'true'
-        env:
-          PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
-          PYPI_USERNAME: __token__
-        run: |
-          poetry publish --build --username $PYPI_USERNAME --password $PYPI_TOKEN
+      - name: download-bdist
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ needs.build.outputs.bdist-package-name }}
+          path: dist
 
-      #----------------------------------------------
-      #    post a message to slack
-      #----------------------------------------------  
+      - name: deploy-to-test-pypi
```

Comment:
I noticed this step only includes deployment to test pypi. Are we no longer deploying to PyPI, or is there a step for PyPI deployment that I'm missing?

Reply:
It publishes to pypi when pre-release is not selected. to test pypi when it is selected. Changing the name now

Reply:
changed in latest commit

Reply:
Thanks for the update!